File: cpp/763.cpp
class Solution {
public:
    vector<int> partitionLabels(string& s) {
        vector<int> res;
        int n = s.length();
        int pi[26] = {0};
        for (int i = n - 1; i >= 0; i--) {
            int t = s[i] - 97;
            if (pi[t] == 0)
                pi[t] = i;
        }
        int pti = 0, cti = 0;
        for (int i = 0; i < n; i++) {
            if (cti < i) {
                res.push_back(cti - pti + 1);
                pti = i;
            }
            int t = pi[s[i] - 97];
            if (t > cti)
                cti = t;
        }
        res.push_back(cti - pti + 1);
        return res;
    }
};

File: python3/763.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        res = []
        pi = [0] * 26
        for i in range(len(s) - 1, -1, -1):
            t = ord(s[i]) - 97
            if pi[t] == 0:
                pi[t] = i
        pti = 0
        cti = 0
        for i in range(0, len(s)):
            if cti < i:
                res.append(cti - pti + 1)
                pti = i
            t = pi[ord(s[i]) - 97]
            if t > cti:
                cti = t
        res.append(cti - pti + 1)
        return res
